SWLOR.Game.Server/Placeable/ControlTower/OnDamaged.cs

```diff
@@ -50,10 +53,20 @@ public bool Run(params object[] args)
             PCBaseStructure structure = _data.Single<PCBaseStructure>(x => x.ID == new Guid(structureID));
             int maxShieldHP = _base.CalculateMaxShieldHP(structure);
             PCBase pcBase = _data.Get<PCBase>(structure.PCBaseID);
+            var playerIDs = _data.Where<PCBasePermission>(x => x.PCBaseID == structure.PCBaseID).Select(s => s.PlayerID);
+            var toNotify = NWModule.Get().Players.Where(x => playerIDs.Contains(x.GlobalID));
+            bool alerted = false;
```

Comment:
This alerted variable seems unnecessary to me. It's always going to be false every time this script runs. Once the script finishes, it'll still be false on the next run.

Moreover, it's going to spam the hell out of players haha. I would look at doing a timestamp (DateTime.UtcNow) and checking against that each time this is fired. If the player hasn't been notified in 15 or 30 seconds (whatever makes sense), then run the following. That way it's immediately apparent there's a problem but they aren't getting spammed about it.